### DISCUSSION TITLE
Change GOMODCACHE

### DISCUSSION THREAD

**Discussion by UncleVic** [golang/vscode-go#1504] [Help] [answered]
Hi there,

How can I change GOMODCACHE?
I don't want installing modules to $HOME/go

I tried to set 
```
    "gopls": {
        "build.env": {
            "GOPATH": "path/go",
            "GOMODCACHE": "path/go/pkg/mod"
        }
    },
```
But it doesn't work :(

Thx

**Comment by wtask** [comment]
Your problem is not vscode-go related, run `go help env`

**Comment by hyangah** [comment]
Try  to set `GOMODCACHE` using `"go.toolsEnvVars"` instead of `gopls` setting. https://github.com/golang/vscode-go/blob/master/docs/settings.md#gotoolsenvvars

**Comment by stamblerre** [comment, marked as answer]
Setting `GOMODCACHE` through the VS Code or `gopls` settings will only affect the value of that environment variable when you are in the editor or the integrated terminal. To set it globally, you will want to run `go env -w GOMODCACHE="path/go/pkg/mod"`.

**Reply by hyangah** [reply]
Right. If you want the change applied globally in the machine, `go env -w` is the way to do so. Also, the `"go.toolsEnvVars"` setting applies only to tools run by the extension, and doesn't apply to the integrated terminal.
